### DISCUSSION TITLE
Override the updated_at

### DISCUSSION THREAD

**Discussion by krichards** [romeerez/orchid-orm#143] [Q&A]
How can I override the updated at when adding and updating rows.

I need to sync with another data source and set the updated at under some circumstances.

I tried adding including the updated_at but it complains that there are multiple instances of the same column.

**Comment by romeerez** [comment]
Could you provide a code example that is causing the error?

Simply providing the `updated_at` key-value to the `update` method should work just fine, so there should be something else that is causing the trouble.

**Reply by krichards** [reply]
So if I use updated_at then the code runs, but it does not use timestamp.  If I use the updatedAt: then I get the error

error: multiple assignments to same column "updated_at"  (I'm using snake case)

When I check the sql generated I can see I get this fior updatedAt
```
  text: 'UPDATE "hid_to_email" SET "updated_at" = $1, "email" = $2, "updated_at" = now() WHERE "hid_to_email"."hid" = $3',
  values: [
    2023-02-02T22:22:22.222Z,
    'audrey_stoltenberg@hotmail.com',
    '3R8K0LDJSFS8'
   ]
```
and this for updated_at

```
  text: 'UPDATE "hid_to_email" SET "email" = $1, "updated_at" = now() WHERE "hid_to_email"."hid" = $2',
  values: [ 'coty_stiedemann@rodrick.ca', 'GGJ5392X4Q4V' ]

```

**Reply by krichards** [reply]
I presume this test should pass? 
 ```
 it('should not update updated_at column when updating snakeCase table with `updated_at = `', () => {
      const db = createDb({
        databaseURL: process.env.PG_URL,
        snakeCase: true,
      });

      const table = db('snake', (t) => ({
        id: t.serial().primaryKey(),
        ...t.timestamps(),
      }));

      const q = table.where().update({ updated_at: "1234"});

      expectSql(
        q.toSql(),
        `
        UPDATE "snake"
        SET "updated_at" = $1
      `,
      );
    });
```

**Reply by romeerez** [reply]
[Here is the test](https://github.com/romeerez/orchid-orm/blob/main/packages/core/src/columns/timestamps.test.ts#L29) to make sure that when you provide your own value for the `updatedAt` it works correctly.

But I can see the problem, it's about using a snake cased `updated_at`.

JS conventions tell us to use `camelCase` in the code; the ORM is designed with this in mind.

By default, columns in the app code are named in `camelCase` and db columns are expected to be in `camelCase`.
You can set `snakeCase: true` ([here are the docs for it](https://orchid-orm.netlify.app/guide/orm-and-query-builder.html#snakecase-option)), columns in the app code are still in `camelCase`, but they are translated to `snake_case` when producing SQL.

If you try to do this:
```ts
const q = table.where().update({ updated_at: "1234"});
```

You should have TS error saying that `updated_at` is not allowed, so if you are doing this and not getting the TS error I'd like to see your code to figure out what went wrong.

**Reply by krichards** [reply]
I think the problem is that the key specified on the snake timestamps is updated_at, whilst the actual key used on the update object will be 'updatedAt' so it is not removed and you get the duplication error.

If you try and use 'updated_at'  it's filtered out because it's not a valid column name and so the default behaviour remains.

**Reply by krichards** [reply]
I've made this change and then my test (using snake case and updatedAt) passes.

**Reply by krichards** [reply]
@romeerez I've created  a PR containing the test and what I believe is the fix.  You should see if the fix is removed you get duplicate updated_at in the SQL.

#144

**Reply by romeerez** [reply]
Thank you a lot for spending time and making a PR! I'll re-check it in the near days, sorry that don't have much time right now, and will merge
